torch/_inductor/decomposition.py

```diff
@@ -357,10 +358,29 @@ def all_dim(input, dim, keepdim=False):
 @register_decomposition(aten.copy)
 def copy(self, src, non_blocking=False):
     intermediate = src.to(self, non_blocking)
-    if self.size() != intermediate.size():
-        return aten.expand_copy.default(intermediate, self.size())
-    else:
+    # cheapest case
+    if self.size() == intermediate.size() and self.stride() == intermediate.stride():
         return intermediate
+    # next cheapest case
+    if self.size() == intermediate.size() and self.is_contiguous():
+        return intermediate.contiguous()
+    # next cheapest case (expand_copy is guaranteed to return a contiguous tensor)
+    if self.is_contiguous():
+        return aten.expand_copy.default(intermediate, self.size())
+    # expensive case
+    # We need to return a tensor with the data of "src",
+    # But with the size/stride/storage_offset of "self" (... and any other metadata! neg, conj, etc).
+    # A problem for another day
+    required_size = compute_required_storage_length(
+        self.size(), self.stride(), self.storage_offset()
+    )
+    out_buffer = torch.empty(required_size, dtype=self.dtype, device=self.device)
+    out_buffer_updated = aten.as_strided_scatter(
```

Comment:
You need to fill in the buffer with the content of self at some point here?

Reply:
hmm, so... do I?

I want to make the argument that this memory should never get read (and the **only** reason we need to allocate it is to keep strides correct so our ops that aren't stride-agnostic are happy, like view).

The example where I think you could imagine that data being accessed is in code like this:
```
a = torch.zeros(2, 2)
a_slice = a[:, 1]
a.copy_(toch.ones_like(a_slice))
# What happens if the user accesses other parts of `a`, that are not `a_slice`?
```

But that case should be fine - any accesses of `a` (or any other views of `a` should be turned into fresh tensors via `slice/select_scatter` call, which will ignore the unitialized data when it's scattering the updated data into the larger tensor.

The other case I can think of is using `as_strided()` to directly access memory outside the bounds of the current slice, but this is banned in functionalization anyway.